src/control/server/instance_storage.go

```diff
@@ -25,6 +24,22 @@ func (ei *EngineInstance) GetStorage() *storage.Provider {
 	return ei.storage
 }
 
+// MountMetadata mounts the configured control metadata location.
+func (ei *EngineInstance) MountMetadata() error {
```

Comment:
Yeah, I followed the path of least resistance on this part. I was also trying to abstract the metadata management a bit, so that we could do the right thing under the covers depending on the config (i.e. MD path or legacy SCM location), so that all the caller cares about is that it's trying to get metadata. I concede that it's a bit weird for the engine to know much about it.